### DISCUSSION TITLE
关于用户手册里的onebot-mirai项目

### DISCUSSION THREAD

**Discussion by pilgrim773** [MrXiaoM/Overflow#34] [常规]
它是一个onebot实现吗？仅用overflow+onebot-mirai应该是不能跑起来的吧？以我浅薄的认识，我感觉它只是一个转发插件，我实际上仍然需要部署并运行一个完整的onebot协议实现对吗？ 
如果是这样，overflow是能直接对接其他onebot生态的实现的，也不需要onebot-mirai来帮助了吧……
如果我说错了，我很抱歉。希望赐教关于onebot-mirai的使用方法。

**Comment by MrXiaoM** [comment]
是的，onebot-mirai 仅仅是把 mirai 变成 Onebot 协议端，让其它 Onebot 应用端连接上去使用 mirai 的协议。
Overflow 的定位是 Onebot 应用端，其出现的目的是在 mirai 协议不可用时，将 mirai 的协议端剔除，将其接口转接到 Onebot 协议端。

所以，你依然需要部署一个 Onebot 协议实现。

**Reply by pilgrim773** [reply]
> onebot-mirai 仅仅是把 mirai 变成 Onebot 协议端，让其它 Onebot 应用端连接上去使用 mirai 的协议

> 所以，你依然需要部署一个 Onebot 协议实现。

我可以这样理解吗：
原本本项目用法是 overflow（onebot 应用端） + XXX（onebot协议实现，如gensoukyo）。
onebot-mirai的通常用法是 XXX（onebot 应用端，如nonebot2）+ onebot-mirai + mirai项目（被转换为onebot协议实现）

但是如果这两者一起使用，由于mirai核心被剔除了，所以缺少了协议实现这一环。

**Reply by MrXiaoM** [reply]
onebot-mirai 仅在可以在 mirai 登录账号时有用，如果 mirai 上面同时使用 Overflow 和 onebot-mirai，那么 。mirai 当前既是 Onebot 应用端又是协议端，协议端会调取 mirai 的接口，而 mirai 的接口又被 Overflow（Onebot 应用端）对接到了其它 Onebot 协议端，此时 onebot-mirai 的作用就是充当中转站。

就跟用户手册里说的一样可以做到 mirai 接龙，只要本地的每一个 mirai 都安装 Overflow 和 onebot-mirai，并且配置好端口，就能把每一个 mirai 都连起来。

当然这么做不是很推荐，协议间转接是有损耗的。

总之，onebot-mirai 这条路基本可以忽略，除非你有账号依然可以稳定在 mirai 使用签名服务登录。

**Reply by pilgrim773** [reply]
qsign已经让我被永封了两个号，我还是探索其他协议实现吧，谢谢回复
